pkg/sdk/logging/model/filter/useragent.go

```diff
@@ -0,0 +1,103 @@
+// Copyright © 2019 Banzai Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package filter
+
+import (
+	"github.com/cisco-open/operator-tools/pkg/secret"
+	"github.com/kube-logging/logging-operator/pkg/sdk/logging/model/types"
+)
+
+// +name:"User Agent"
+// +weight:"200"
+type _hugoUserAgent interface{} //nolint:deadcode,unused
+
+// +docName:"Fluentd UserAgent filter"
+// Fluentd Filter plugin to parse user-agent
+// More information at https://github.com/bungoume/fluent-plugin-ua-parser
+type _docUserAgent interface{} //nolint:deadcode,unused
+
+// +name:"UserAgent"
+// +url:"https://github.com/bungoume/fluent-plugin-ua-parser"
+// +version:"1.2.0"
+// +description:"Fluentd UserAgent filter"
+// +status:"GA"
+type _metaUserAgent interface{} //nolint:deadcode,unused
+
+// +kubebuilder:object:generate=true
+type UserAgent struct {
+	//Target key name (default: user_agent)
+	KeyName string `json:"key_name,omitempty"`
+	//Delete input key (default: false)
+	DeleteKey bool `json:"delete_key,omitempty"`
+	//Output prefix key name (default: ua)
+	OutKey string `json:"out_key,omitempty"`
+	//Join hashed data by '_' (default: false)
+	Flatten bool `json:"flatten,omitempty"`
+}
+
+// ## Example `UserAgent` filter configurations
+// ```yaml
+// apiVersion: logging.banzaicloud.io/v1beta1
+// kind: Flow
+// metadata:
+//
+//	name: demo-flow
+//
+// spec:
+//
+//		filters:
+//		  - useragent:
+//		      key_name: my_agent
+//	        delete_key: true
+//	        out_key: ua_fields
+//	        flatten: true
+//		selectors: {}
+//		localOutputRefs:
+//		  - demo-output
+//
+// ```
+//
+// #### Fluentd Config Result
+// ```yaml
+// <filter **>
+//
+//		@type ua_parser
+//		@id test_useragent
+//	 key_name my_agent
+//	 delete_key true
+//	 out_key ua_fields
+//	 flatten true
+//
+// </filter>
+// ```
```

Comment:
please follow multiline formatting pattern based on this example:
https://github.com/kube-logging/logging-operator/blob/b4191ebb4d856b64cc5b6600d02e22c357b4a586/pkg/sdk/logging/model/syslogng/output/s3.go#L25

Reply:
If yes, would you like me to do that for all the filters in a separate PR?

Reply:
It seems we haven't fixed all the bad patterns yet. You don't have to fix all the other filters, but let's format this one properly.

Follow up: https://github.com/kube-logging/logging-operator/issues/1596

Reply:
Ok, I don't know if what I did is what you meant, but I did the best I could figure.

Reply:
thanks, I've made a few updates to the formatting, but seems to be good now!